### ISSUE TITLE
Reply to issue

### ISSUE THREAD

**Issue by vmasanas** [dubeaud/bugnet#89]
How difficult would be to implement a "reply to" feature so that you could reply to an email notification and get the reply posted as a comment? I've used this in basecamp (for example) and it's a pretty nice thing.
I was thinking on adding a token on the body or something similar that could be used when parsing the email later on. But this doesn't seem very reliable. In basecamp they use a custom email per todo, like todo-1333r32605-51werwe14a08348f2ec774f@basecamp.com

**Comment by dubeaud** [comment]
I don't think it would be very difficult but I think you would want a separate service, or instance of the mailbox reader configured for a global mailbox to work with BugNET to parse notification emails.  Here is some discussion about the email address being the best way to do this.   

http://stackoverflow.com/questions/9494542/how-can-i-know-which-email-is-a-reply-to-another-email-in-c

I do wonder how basecamp is constructing the token, is it an encrypted user id and issue id. Simple way would just be plain text like stack overflow.

**Comment by vmasanas** [comment]
Maybe an even simple solution would be to add the ticket id onto the subject line and try to parse based on that. I mean, you get a notification with a special token on the subject. Given you don't change this on the reply, could be an easy way to identify the source ticket.

**Comment by dubeaud** [comment]
I think we would need the user id and the ticket id.  I suppose to you could look up the user by the email id potentially but might make it easier to parse out of the email subject or replyto address.

**Comment by vmasanas** [comment]
I've implemented this using an option most email servers support: plus addressing. See here for example: http://gmailblog.blogspot.com.es/2008/03/2-hidden-ways-to-get-more-from-your.html
It's comming on a new pull request. Take a look at let me know.

**Comment by vmasanas** [comment]
Regarding this point, which is already sent in a Pull Request, I'm not sure how to improve the reply to so not all text from the email gets injected into the comment.
The point is that when the user uses the "Reply to" button in the email client, it will include the text for the notification, so the comment on the issue will contain too many stuff that ideally shouldn't be there.
I'm now manually removing all the text on the email before replying but this is far from being the best option. Any idea on how to do this?

**Comment by dubeaud** [comment]
I have seen other issue trackers use a "----- REPLY ABOVE THIS LINE" type of syntax, that way you can ignore everything below.  Something like this? https://www.redmine.org/attachments/1947/redmine-0.8.3-strip_quoted_top_and_bottom.patch

**Comment by vmasanas** [comment]
(this comment should come from a direct reply from outlook. Let me see how it appears on the web. )

I suppose my email signature and previous message will appear there.


Vicenç Masanas
Tel: +34 653 391 064 - Skype: vmasanas - @:  <mailto:vmasanas@disgrafic.com> vmasanas@disgrafic.com 


C/ Sant Mer, 10 - 17820 Banyoles – Girona - SPAIN
 <http://www.disgrafic.com/> www.disgrafic.com

 

De: Davin Dubeau [mailto:notifications@github.com] 
Enviado el: dimecres, 25 de març de 2015 0:29
Para: dubeaud/bugnet
CC: Vicenç Masanas
Asunto: Re: [bugnet] Reply to issue (#89)

 

I have seen other issue trackers use a "----- REPLY ABOVE THIS LINE" type of syntax, that way you can ignore everything below. Something like this? https://www.redmine.org/attachments/1947/redmine-0.8.3-strip_quoted_top_and_bottom.patch

**Comment by vmasanas** [comment]
Ok, seems they dont' remove anything... not good.
On you suggestion for the ---- Reply above.... yes that's something I've also seen in Basecamp. What I don't like about this approach is that it still will include the email signature and the headers for the previous email.

**Comment by vmasanas** [comment]
BTW, I've had all the features I submitted on my last pull request running on production this week and the seem to work fine !

**Comment by dubeaud** [comment]
Great, I will merge the code

**Comment by vmasanas** [comment]
Let me  review the comment feature this weekend. Maybe I can incorporate the option to strip the previous email text.

**Comment by dubeaud** [comment]
Sounds good.
